app/kubemci/pkg/gcp/healthcheck/healthchecksyncer_test.go

```diff
@@ -144,7 +238,7 @@ func TestDeleteHealthCheck(t *testing.T) {
 			Protocol: utils.ProtocolHTTP,
 		},
 	}
-	if _, err := hcs.EnsureHealthCheck(lbName, ports, clientfake.NewSimpleClientset(), false); err != nil {
+	if _, err := hcs.EnsureHealthCheck(lbName, ports, &clientfake.Clientset{}, false); err != nil {
```

Comment:
Its not clear to me, honestly, when should I use NewSimpleClientset and when should I instantiate Clientset directly.

Curious to know your motivation for this change. Is this change required for the test to pass?

Reply:
At first I had problems with `NewSimpleClientSet` because I didn't know that it adds a catch-all reactor, so if you want to add a reactor that has precedence you need to use `PrependReactor` instead of `AddReactor`. I thought it would be more clear what was going on if we just instantiated Clientset and explicitly set the reactors we need. I think `NewSimpleClientset` is useful when you're just unit-testing one call and you don't care about what call is made, just that it returns a specific object.